### ISSUE TITLE
Unable to assign columns using (names(.SD)) :=

### ISSUE THREAD

**Issue by skanskan** [Rdatatable/data.table#4905]
Hello.

I want to modify the columns containing a given pattern in their names.

Say I have this toy example:
`myDT <- data.table(cod1=1:20, cod2=20:1, other1=rnorm(20), other2=rnorm(20))`

And I want  to convert the content of the columns whose name start by "cod" to factors.

With dplyr I can use this compact notation:
`myDT %>% mutate(across(contains("cod"),as.factor)) 
`

With data.table I would use this:
```
cols <- grep("cod", names(myDT), value = T)
myDT[,(cols):= lapply(.SD, as.factor), .SDcols = cols] 
```

It's much longer and it generates new variables we don't need anymore.
Is there any shorter/simpler/faster alternative?

I have tried this:
`myDT[, (names(.SD)) := lapply(.SD, as.factor) , .SDcols = patterns("cod")]
`
but it doesn't work.

**Comment by jangorecki** [comment]
There is FR for that, and maybe even PR as well.

**Comment by MichaelChirico** [comment]
duplicate of #795, will be fixed in #4163
